lib/system/supports-colors.js

```diff
@@ -31,9 +31,8 @@ module.exports = (function () {
     return false;
   }
 
-  if (argv.indexOf('--color') !== -1 ||
-    argv.indexOf('--color=true') !== -1 ||
-    argv.indexOf('--color=always') !== -1) {
+  if (!argv.indexOf('--no-color') !== -1 ||
+    !argv.indexOf('--color=false') !== -1) {
```

Comment:
This is wrong for multiple reasons.

1. The `||` is an or, which means this will always enable color when either the first _or_ the second argument is missing. This is _always_ the case, except when you specify _both_ arguments, which does not make much sense. I assume this was meant to be an `&&`.
2. With an `&&` this is the exact inverse of the `if` above. Which means all the code below this second `if` can never be reached.

Look at the code below. The issue you are experiencing may be hidden there.

I suggest to close this patch.